### PR TITLE
lab: Fix inconsistencies in chapter introductions

### PR DESCRIPTION
**Pull request by teodorjuravlea**
This is the fix I've suggested for #107.
I haven't been able to setup the Docker container to test the changes locally, so please check if I've changed everything correctly.
The 'Compute' lab chapter could use an introduction paragraph.

### CODE REVIEW

File: content/lab-setup/lab-setup.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Setting up the Lab environment
+
+If you have already cloned the repository, make sure it is updated.
+Run this command inside the repository clone directory:
```
Comment (Alex-deVis) [inline content/lab-setup/lab-setup.md:4]:
```suggestion
```

File: content/lab-setup/lab-setup.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+```console
+git pull --rebase
```
Comment (Alex-deVis) [inline content/lab-setup/lab-setup.md:7]:
```suggestion
cd operating-systems
git pull --rebase
```

**Review by Alex-deVis — CHANGES_REQUESTED** [review]
I left a few nitpicks. Besides those, the path to `lab-setup.md` is wrong.

I think its overkill to name the section `lab-setup` and add a file with the same name. Let us call the directory `misc/`. @teodutu, please provide your pedantic view on this.

**Comment by teodorjuravlea** [comment]
Thank you. I've added your improvements, and I've also renamed the directory to `misc/`, since I assume that has to change regardless.
Are you sure the path to `lab-setup` in `config.yaml` is incorrect? It seems correct to me, based on how the other sections are set.

File: config.yaml [hunk 3]
```diff
@@ -55,14 +55,14 @@ docusaurus:
...
-        - Application-Interaction:
+        - Application Interaction:
             name: Application-Interaction
             location: /build/embed_reveal/Application-Interaction
       - Lab:
-        - Software-Stack:
-          - Introduction:
-              location: content/chapters/software-stack/lab
-              name: content/introduction
+        - Setting up the Lab environment:
```
Comment (teodutu) [inline config.yaml:62]:
```suggestion
        - Setting up the Lab Environment:
```

File: content/misc/lab-setup.md [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+If you have already cloned the repository, make sure it is updated:
+
+```console
+cd operating-systems
+git pull --rebase
+```
+
+The command may fail if you have uncommitted changes.
+If that is the case, commit changes and retry.
```
Comment (teodutu) [inline content/misc/lab-setup.md:11]: Commit or stash and pop after `git pull`? I generally support commits, but last semester's experience showed me that maybe the simpler option is the best / fastest. @Alex-deVis, what do you say?

File: content/misc/lab-setup.md [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+If you haven't already cloned the repository, do it and enter the repository:
```
Comment (teodutu) [inline content/misc/lab-setup.md:13]:
```suggestion
If you haven't already cloned the repository, do so and then enter the repository:
```

File: content/misc/lab-setup.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+```console
+git clone https://github.com/open-education-hub/operating-systems
+cd operating-systems
+```
+
+Navigate to the chapter lab directory:
+
+```console
+cd content/chapters/software-stack/lab/
+```
```
Comment (teodutu) [inline content/misc/lab-setup.md:24]: The Docker container defines some aliases to navigate more quickly [1]. I still think they're pointless, but maybe we should mention them here. @Alex-deVis would love this.

[1] https://github.com/open-education-hub/operating-systems/blob/b6a0cbdff828bdabfb7403a597e325c747e6bfbe/util/os-runner/config_terminal.sh#L21-L25

**Review by teodutu — CHANGES_REQUESTED** [review]
Great changes! The path to the setup guide in `config.yaml` looks right. Address my comments and then we can merge this.

File: content/misc/lab-setup.md (hunk 4 above)
Comment (Alex-deVis) [reply content/misc/lab-setup.md:11]: Right, but only say `git stash` -> `git pull` -> `git stash pop`. Give one comprehensive option, not a bunch.

File: config.yaml [hunk 7]
```diff
@@ -55,14 +55,14 @@ docusaurus:
...
+            location: content/misc
+            name: lab-setup
```
Comment (Alex-deVis) [inline config.yaml:64]: > Are you sure the path to lab-setup in config.yaml is incorrect? It seems correct to me, based on how the other sections are set.

`location` + `name` determine the path to the file. I think one example would solve this:

Software Stack: `locaction = content/chapters/software-stack/lab` and `name = content/overview`. If you look for the overview file's path in Software Stack is `content/chapters/software-stack/lab/content/overview.md`.

Using the same logic for lab-env: `location = content/misc` and `name = lab-setup` leads to the path `content/misc/lab-setup.md`, which is our file.

File: content/misc/lab-setup.md (hunk 4 above)
Comment (teodutu) [reply content/misc/lab-setup.md:11]: This will confuse people. I wouldn't give options. One way to rule them all.

Comment (Alex-deVis) [reply content/misc/lab-setup.md:11]: To sort this out add the inline suggestion and mark this as resolved.

**Comment by teodorjuravlea** [comment]
I've added the changes, mentioned all possible chapters to navigate to, and I've added the mention about aliases in a new commit. I've copied the alias descriptions directly from `config_terminal.sh`, so tell me if I should change its structure.
I'll add the info about uncommited changes as soon as you decide on the final version.

File: content/misc/lab-setup.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+
+If you have already cloned the repository, make sure it is updated:
+
+```console
+cd operating-systems
+git pull --rebase
+```
+
+The command may fail if you have uncommitted changes.
+If that is the case, commit changes and retry.
```
Comment (teodutu) [inline content/misc/lab-setup.md:11]:
```suggestion
If that is the case, stash your changes, retry, and pop the stash:

```console
student@os:~/operating-systems$ git stash

student@os:~/operating-systems$ git pull --rebase

student@os:~/operating-systems$ git stash pop
```
```

The suggestion is a bit malformed due to the backticks interfering with each other.

File: content/misc/lab-setup.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+If you haven't already cloned the repository, do so and then enter the repository:
+
+```console
+git clone https://github.com/open-education-hub/operating-systems
+cd operating-systems
```
Comment (teodutu) [inline content/misc/lab-setup.md:17]:
```suggestion
student@os:~$ git clone https://github.com/open-education-hub/operating-systems

student@os:~$ cd operating-systems
```

File: content/misc/lab-setup.md [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+Navigate to a chapter's lab directory:
+
+```console
+cd content/chapters/<chapter-name>/lab/
```
Comment (teodutu) [inline content/misc/lab-setup.md:23]:
```suggestion
student@os:~/operating-systems$ cd content/chapters/<chapter-name>/lab/
```

File: content/misc/lab-setup.md [hunk 11]
```diff
@@ -0,0 +1,34 @@
+# Setting up the Lab Environment
+
+If you have already cloned the repository, make sure it is updated:
+
+```console
+cd operating-systems
+git pull --rebase
```
Comment (teodutu) [inline content/misc/lab-setup.md:7]:
```suggestion
student@os:~$ cd operating-systems

student@os:~/operating-systems$ git pull --rebase
```
It's better to leave an empty line between subsequent commands to make them more clearly distinguishable.

**Review by teodutu — CHANGES_REQUESTED** [review]
I made suggestions about stashing changes and added a prompt to console snippets.

**Review by teodutu — APPROVED** [review]
:+1:

**Review by Alex-deVis — APPROVED** [review]

**Pull request merged**